pkg/pkg_test.go

```diff
@@ -6,34 +6,117 @@ import (
+			Expect(machineType[CpuCores]).To(Equal("48"))
 		})
 		It("Convert parsed resource into yaml", func() {
 			err := convert.ResourcesToYamlMachineTypes(machineTypes, outputFile)
-			Expect(err).To(BeNil())
+			Expect(err).ToNot(HaveOccurred())
 			testFile, err := os.ReadFile(outputFile)
 
+			Expect(err).ToNot(HaveOccurred())
 			Expect(testFile).To(ContainSubstring(machineTypeId))
-			Expect(testFile).To(ContainSubstring("12xlarge"))
-			Expect(testFile).To(ContainSubstring("48"))
+			Expect(string(testFile)).To(Equal(testOutputFile))
+		})
+		It("Test output file creation", func() {
+			err := convert.ResourcesToYamlMachineTypes(machineTypes, "output/test.configmap.yaml")
+			Expect(err).ToNot(HaveOccurred())
+			file, err := os.ReadFile("output/test.configmap.yaml")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(file)).To(Equal(testOutputFile))
+			err = os.Remove("output/test.configmap.yaml")
+			Expect(err).ToNot(HaveOccurred())
+			err = os.Remove("output")
+			Expect(err).ToNot(HaveOccurred())
+			// File creation should only happen for the 'default' option (same dir, inside a dir called 'output')
+			err = convert.ResourcesToYamlMachineTypes(machineTypes, "otpt/test.configmap.yaml")
+			Expect(err).To(HaveOccurred())
+		})
+		It("Fail for each field supplied with the wrong data type", func() {
+			// Check function which is to be used by the parse command
+			machineTypes, err := convert.CsvToResources("test/badDataTypes.csv")
+			Expect(err).To(HaveOccurred())
+			Expect(machineTypes).To(BeNil())
+			// Check each specific field validation
+			machineTypes, err = convert.CsvToResources("test/test.csv")
+			Expect(err).ToNot(HaveOccurred())
+			machineType := machineTypes[0]
+			machineTypeId := machineType[Id].(string)
+			field := GenericName
+			Expect(makeFakeMachineTypeDataType(machineType, field, 1234).Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = SizePretty
+			Expect(makeFakeMachineTypeDataType(machineType, field, 1234).Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = Category
+			Expect(makeFakeMachineTypeDataType(machineType, field, 1234).Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = Memory
+			Expect(makeFakeMachineTypeDataType(machineType, field, "aaa").Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = CpuCores
+			Expect(makeFakeMachineTypeDataType(machineType, field, "aaa").Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = CloudProviderId
+			Expect(makeFakeMachineTypeDataType(machineType, field, 1234).Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
+			field = Name
+			Expect(makeFakeMachineTypeDataType(machineType, field, 1234).Error()).
+				To(Equal(fakeDataTypeValidationError(machineTypeId, field).Error()))
```

Comment:
Why are we using the `field` variable instead of directly using the constant (`GenericName`, `SizePretty`, etc)?

Reply:
ah yeah, that is better, this was left over from before they were consts, will change